ramanujantools/cmf/cmf_test.py

```diff
@@ -42,29 +42,19 @@ def test_trajectory_matrix_diagonal():
 
 def test_back_negates_forward():
     cmf = known_cmfs.e()
-    assert (
-        Matrix.eye(2) == (cmf.M(x, True) * cmf.M(x, False).subs({x: x + 1})).reduce()
-    )
-    assert (
-        Matrix.eye(2) == (cmf.M(x, False) * cmf.M(x, True).subs({x: x - 1})).reduce()
-    )
-    assert (
-        Matrix.eye(2) == (cmf.M(y, True) * cmf.M(y, False).subs({y: y + 1})).reduce()
-    )
-    assert (
-        Matrix.eye(2) == (cmf.M(y, False) * cmf.M(y, True).subs({y: y - 1})).reduce()
-    )
+    assert Matrix.eye(2) == (cmf.M(x, True) * cmf.M(x, False).subs({x: x + 1})).reduce()
+    assert Matrix.eye(2) == (cmf.M(x, False) * cmf.M(x, True).subs({x: x - 1})).reduce()
+    assert Matrix.eye(2) == (cmf.M(y, True) * cmf.M(y, False).subs({y: y + 1})).reduce()
+    assert Matrix.eye(2) == (cmf.M(y, False) * cmf.M(y, True).subs({y: y - 1})).reduce()
 
 
 def test_trajectory_matrix_negative_axis():
     cmf = known_cmfs.e()
-    assert (
-        cmf.trajectory_matrix({x: -3, y: 0})
-        == cmf.M(x, False).walk({x: -1, y: 0}, 3, {x: x, y: y}).reduce()
+    assert cmf.trajectory_matrix({x: -3, y: 0}) == cmf.M(x, False).walk(
+        {x: -1, y: 0}, 3, {x: x, y: y}
     )
-    assert (
-        cmf.trajectory_matrix({x: 0, y: -2})
-        == cmf.M(y, False).walk({x: 0, y: -1}, 2, {x: x, y: y}).reduce()
+    assert cmf.trajectory_matrix({x: 0, y: -2}) == cmf.M(y, False).walk(
+        {x: 0, y: -1}, 2, {x: x, y: y}
     )
 
```

Comment:
This should be correct since you removed the `.reduce()` from the definition of trajectory matrix.